File: testbed/test/cruise/associations/ReflexiveOneTest.java
package cruise.associations;

import org.junit.*;

import cruise.associations.MentorAF;

public class ReflexiveOneTest
{

  @Test
  public void getFriend()
  {
    MentorAF m = new MentorAF("m1","m2");
    MentorAF m2 = m.getFriend();
    
    Assert.assertEquals(m2, m.getFriend());
    Assert.assertEquals(m, m2.getFriend());
  }
  
}

File: testbed_python/test/associations/reflexive_one_test.py
import unittest

# Test classes are imported dynamically
# Must be generated into ImportModules' namespace then imported
from ImportModules import importModules

importModules(["MentorAF"], ["associations"])
from ImportModules import *

class ReflexiveOneTest(unittest.TestCase):
    def test_getFriend(self):
        m = MentorAF.MentorAF("m1","m2")
        m2 = m.getFriend()
    
        self.assertEqual(m2, m.getFriend())
        self.assertEqual(m, m2.getFriend())